care/emr/resources/scheduling/schedule/spec.py

```diff
@@ -58,37 +58,59 @@ def validate_availability(cls, availabilities: list[AvailabilityDateTimeSpec]):
 
 class AvailabilityUpdateSpec(AvailabilityBaseSpec):
     name: str
-    tokens_per_slot: int = Field(ge=1)
+    tokens_per_slot: int | None = Field(ge=1)
     reason: str = ""
 
     def perform_extra_deserialization(self, is_update, obj):
         old_instance = Availability.objects.get(id=obj.id)
 
-        future_slots = TokenSlot.objects.filter(
-            availability__id=obj.id,
-            start_datetime__gte=timezone.now(),
-        )
-
-        # Prevents editing tokens per slot below the max allocated tokens of future slots
-        if old_instance.tokens_per_slot != self.tokens_per_slot:
-            max_allocated = future_slots.aggregate(max=Max("allocated"))["max"] or 0
-            if max_allocated > self.tokens_per_slot:
-                msg = (
-                    "Cannot modify tokens per slot as it would exclude some allocated slots. "
-                    f"Max allocated tokens is {max_allocated} while new tokens per slot is {self.tokens_per_slot}."
+        if old_instance.slot_type == "appointment":
```

Comment:
Isnt it easier to not allow edits on appointment type? we can ask them to delete and recreate, changing type is an edge case in our logic and we really dont want all this logic for something the user will never use.